diff --git a/src/internal/server/index.js b/src/internal/server/index.js
index 6098b496c5acbad719c34b0a036fc8451b77c121..7da20e493e47b693f32576ad8b4681fa2b63a9d4 100644
--- a/src/internal/server/index.js
+++ b/src/internal/server/index.js
@@ -105,57 +105,16 @@ function props_id_generator(prefix) {
  * @returns {RenderOutput}
  */
 export function render(component, options = {}) {
-	const uid = props_id_generator(options.idPrefix ? options.idPrefix + '-' : '');
-	/** @type {Payload} */
-	const payload = {
-		out: '',
-		css: new Set(),
-		head: { title: '', out: '', css: new Set(), uid },
-		uid
-	};
-
-	const prev_on_destroy = on_destroy;
-	on_destroy = [];
-	payload.out += BLOCK_OPEN;
-
-	let reset_reset_element;
-
-	if (DEV) {
-		// prevent parent/child element state being corrupted by a bad render
-		reset_reset_element = reset_elements();
-	}
-
-	if (options.context) {
-		push();
-		/** @type {Component} */ (current_component).c = options.context;
-	}
-
-	// @ts-expect-error
-	component(payload, options.props ?? {}, {}, {});
+	const uid = (() => {
+		var pfx = options.idPrefix ?? 's';
+		var uid = 0;
+		return () => `${pfx}${uid++}`;
+	})();
 
-	if (options.context) {
-		pop();
-	}
-
-	if (reset_reset_element) {
-		reset_reset_element();
-	}
-
-	payload.out += BLOCK_CLOSE;
-	for (const cleanup of on_destroy) cleanup();
-	on_destroy = prev_on_destroy;
-
-	let head = payload.head.out + payload.head.title;
+	const payload = { out: '', uid };
+	component(payload, options.props ?? {});
 
-	for (const { hash, code } of payload.css) {
-		head += `<style id="${hash}">${code}</style>`;
-	}
-
-	return {
-		head,
-		html: payload.out,
-		body: payload.out
-	};
+	return { body: payload.out };
 }
 
 /**
@@ -493,12 +452,7 @@ export { await_block as await };
 
 /** @param {any} array_like_or_iterator */
 export function ensure_array_like(array_like_or_iterator) {
-	if (array_like_or_iterator) {
-		return array_like_or_iterator.length !== undefined
-			? array_like_or_iterator
-			: Array.from(array_like_or_iterator);
-	}
-	return [];
+	return array_like_or_iterator;
 }
 
 /**
diff --git a/src/internal/shared/attributes.js b/src/internal/shared/attributes.js
index c8758c1d4d4d68cfe3a3ee76c77605ae65285841..92a352784ca26c3404c86294621f8baf06fd2bab 100644
--- a/src/internal/shared/attributes.js
+++ b/src/internal/shared/attributes.js
@@ -23,7 +23,7 @@ const replacements = {
  */
 export function attr(name, value, is_boolean = false) {
 	if (value == null || (!value && is_boolean)) return '';
-	const normalized = (name in replacements && replacements[name].get(value)) || value;
+	const normalized = value;
 	const assignment = is_boolean ? '' : `="${escape_html(normalized, true)}"`;
 	return ` ${name}${assignment}`;
 }
